web-filter-base/src/com/untangle/app/web_filter/WebFilterDecisionEngine.java

```diff
@@ -139,7 +141,16 @@ public HttpRedirect checkRequest(AppTCPSession sess, InetAddress clientIp, int p
                      */
                     if (matcherO == null || !(matcherO instanceof GlobMatcher)) {
                         try{
-                            matcher = GlobMatcher.getMatcher("*\\b" + rule.getString() + "\\b*");
+                            //Code to handle special characters in search terms
+                            String match = null;
+                            Pattern specialCharacter = Pattern.compile(SPECIAL_CHARACTERS);
```

Comment:
Actually, create a "Pattern SPECIAL_CHARACTER" static variable like the SEPARATORS_REGEX.